packages/lab/src/date-picker/useFocusOut.ts

```diff
@@ -0,0 +1,65 @@
+import type { ElementProps, FloatingContext } from "@floating-ui/react";
+import { useMemo } from "react";
+
+export interface UseFocusOutProps {
+  /**
+   * Whether the hook is enabled
+   * @default true
+   */
+  enabled?: boolean;
+}
+
+function getTabbableElements(parent: Element): HTMLElement[] {
+  const focusableSelectors = [
+    "button:not([disabled])",
+    "input:not([disabled])",
+    '[tabindex]:not([tabindex="-1"])',
+  ];
+  const tabbableElements = parent.querySelectorAll<HTMLElement>(
+    focusableSelectors.join(","),
+  );
+  return Array.from(tabbableElements).filter(
+    (element) => !element.hasAttribute("disabled") && element.tabIndex >= 0,
+  );
+}
+//
+// This hook is needed to close the overlay, when,
+// 1. the floating element is opened
+// 2. the user re-focuses into the reference element with the mouse
+// 3. the user tabs out of the reference element
+// Without this hook, the floating element can re-receive focus
+export function useFocusOut(
```

Comment:
Not sure whether it's code here, if i 
- open calendar via icon
- mouse click input to focus input (panel open)
- tab to focus on calendar icon (panel open)
- shift + tab to focus back to input (panel is closed)
Should panel be closed when focus within date picker? Not sure what will happen if there is additional adornment

Reply:
will take that as a bug... I think it should stay open.